Fix GROMACS using -x to pass environment variables during mpirun, which is only
supported for OpenMPI, not for MPICH. For MPICH, use -env instead.

Requires 'GROMACS-2023.1_fix_tests_for_gmx_thread_mpi.patch' to be applied first.

Author: Jan Reuter (JSC)

diff --color -Naur gromacs-2025.2.orig/python_packaging/sample_restraint/tests/CMakeGROMACS.txt gromacs-2025.2/python_packaging/sample_restraint/tests/CMakeGROMACS.txt
--- gromacs-2025.2.orig/python_packaging/sample_restraint/tests/CMakeGROMACS.txt	2025-12-09 20:55:21.376420416 +0100
+++ gromacs-2025.2/python_packaging/sample_restraint/tests/CMakeGROMACS.txt	2025-12-09 20:54:40.795568662 +0100
@@ -61,7 +61,8 @@
     #    (https://www.open-mpi.org/doc/v3.0/man1/mpiexec.1.php)
     add_custom_target(
         gmxapi_extension_pytest_mpi
-        COMMAND ${MPIEXEC_EXECUTABLE} -n 4 -x PYTHONPATH=$ENV{PYTHONPATH}:${GMXAPI_PYTHON_STAGING_DIR}:${PLUGINPATH}
+        # MPICH uses -env instead of -x to pass environment variables
+        COMMAND ${MPIEXEC_EXECUTABLE} -n 4 -env PYTHONPATH $ENV{PYTHONPATH}:${GMXAPI_PYTHON_STAGING_DIR}:${PLUGINPATH}
             ${PYTHON_EXECUTABLE} -m mpi4py -m pytest -x --log-cli-level ERROR ${CMAKE_CURRENT_SOURCE_DIR}
             DEPENDS gmxapi_extension _gmxapi gmx
             )
